dali/CMakeLists.txt

```diff
@@ -70,6 +70,20 @@ install(TARGETS ${dali_lib} DESTINATION lib)
 install(DIRECTORY ${CMAKE_CURRENT_LIST_DIR} DESTINATION
   include FILES_MATCHING PATTERN "*.h" PATTERN "*_test.h" EXCLUDE)
 
+# install libdali to build dir
+add_custom_command(
+    TARGET ${dali_lib} POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy
+            $<TARGET_FILE:${dali_lib}>
+            ${CMAKE_BINARY_DIR}/lib/$<TARGET_FILE_NAME:${dali_lib}>)
+
+# filter all *.h which don't have 'test' in name, create pairs 'src dst' string and call xargs with install command
+# to copy with dest recursive dir creation
+add_custom_command(
+    TARGET ${dali_lib} POST_BUILD
+    COMMAND find ${CMAKE_CURRENT_LIST_DIR} -regex ".*\\\\.h" | grep --invert test
+    | sed -r 's,\(^${CMAKE_CURRENT_LIST_DIR}\)\(.*\),\\1\\2 ${CMAKE_BINARY_DIR}/include/dali\\2,'
+    | xargs -n 2 install -D)
```

Comment:
This is fairly magical :)  Not wrong, but, well, I'm glad it's commented.

Reply:
I don't have any better idea how to do that in a shorter and cleaner form. Any suggestions are welcome.